core/src/components/picker-column/picker-column.tsx

```diff
@@ -132,6 +141,16 @@ export class PickerColumnCmp implements ComponentInterface {
     const scaleStr = `scale(${this.scaleFactor})`;
 
     const children = this.optsEl.children;
+
+    // Options might have been changed
+    // ex: adding or removing
+    // `this.optsEl` children will not match the new `this.col` order
+    // until after `componentDidUpdate()`
+    // Skip this update cycle
+    if (children.length !== col.options.length) {
+      return;
+    }
```

Comment:
This happen when using multiple columns:
1. Recommended: Add a `console.log()` inside the if statement
2. Navigate to the [test page](http://localhost:3333/src/components/picker-column/test/dynamic-options)
3. Click the "Remove option" under the multiple columns
4. Select the last option ("Artichoke") on the second/right column
5. Select the second option ("Responsive") on the first/left column

Without this if statement then the app will throw this error: `TypeError: opt is undefined`

Reply:
This is happening because we are still calling `this.refresh` from the `colChanged` watch callback. If we only want to run `this.refresh` once the component has re-rendered, then we probably should not be calling `this.refresh` in `colChanged` anymore. Removing that means we can also remove the above ^ check.